lib/membrane/rtp/twcc_sender/congestion_control.ex

```diff
@@ -51,16 +51,10 @@ defmodule Membrane.RTP.TWCCSender.CongestionControl do
     a_hat: 300_000.0,
     # sender-side bandwidth estimation in bps
     as_hat: 300_000.0,
-    # latest estimates of receiver-side bandwidth
-    r_hats: [],
-    # time window for measuring the received bitrate, between [0.5, 1]s (reffered to as "T" in the RFC)
-    target_receive_interval: Time.milliseconds(750),
-    # accumulator for packet sizes in bits that have been received through target_receive_interval
-    packet_received_sizes: 0,
-    # starting timestamp for current packet received interval
-    packet_received_interval_start: nil,
-    # last timestamp for current packet received interval
-    packet_received_interval_end: nil,
+    # latest receiver-side incoming bitrates when we were in the decrease state
+    decrease_r_hats: [],
```

Comment:
This could also be `:queue`, you always add one element at head, but then use `Enum.take` to drop the last element which is not very efficient

Reply:
Good point but
1. I would have to store queue size on my own as it is needed in line 303
2. the way we calculate variance in `std_dev/1` isn't the most correct one as we might get catastophic cancellation 
3. exp moving avg could be calculated in constant time but to do so we need to resolve the 2nd point at first as variance can probably be calculated in constant time too
4. I found constant time implementation in Pion https://github.com/pion/interceptor/blob/master/pkg/gcc/rate_controller.go#L33-L48 but honestly speaking I don't fully get how they calculate variance

To sum up,  this code needs to be refactored a little bit anyway so maybe we could do this in a separate PR and don't introduce queue at the moment :stuck_out_tongue:

Reply:
@Observer46 helped me a lot and we found similar implementation in Chromium

https://source.chromium.org/chromium/chromium/src/+/main:third_party/webrtc/modules/congestion_controller/goog_cc/link_capacity_estimator.cc;drc=27e39fa1dd71076618c358639ed8a327bc3873c4;bpv=0;bpt=1;l=45